package/toltec-bootstrap/toltecctl

```diff
@@ -806,6 +808,14 @@ uninstall() {
     }
     set -o functrace
     trap 'handle-error ${LINENO}' ERR
+    handle-exit() {
+        if ! $completed && $success && $clean; then
```

Comment:
completed vs complete, wrong variable names

Reply:
You seem to be reviewing an outdated version, this is `$complete` for me when I look

Reply:
Ok, I'm reviewing patches in order, not a full diff. I'm more used to a workflow where a PR is force updated instead of incremental patches and a squash afterwards.

Reply:
Also I think there is operator order issue here:

$ export VAR1=true
$ export VAR2=true
$ export VAR3=true
$ if ! $VAR1 && $VAR2 && $VAR3; then echo failed; fi
$ export VAR2=false
$ if ! $VAR1 && $VAR2 && $VAR3; then echo failed; fi
$ if ! ($VAR1 && $VAR2 && $VAR3); then echo failed; fi
failed

Reply:
Since they are declared as local it could be that they aren't working as expected. Although when I tested this originally it was working as I expected.

Reply:
I don't think local makes any difference here, `!` has higher priority than `&&` hence it applies only to VAR1 and from the look of it you wanted to apply it to the whole expression (V1 && V2 && V3)

Reply:
I would have wrapped it in `(` and `)` if I was trying to apply it to all of them, I'm checking if not complete, and success and clean. success/clean are true by default and get changed to false when something else handles reporting an error.

Reply:
In this case, I'm only wanting to handle the error if the error handler is triggered and we did not complete the uninstall, and nothing set success or clean to false due to their own error handling.

It is less than ideal, thus the extra comments on the variable declarations. I couldn't really come up with a cleaner way to handle this with the limitations bash has on error handling.

Reply:
I would have to analyze this fully to check the logic completely. I'm too tired now.

What I did find though is that you make a dep between `clean` and `success` so checking both in this line might be at least redundant.

https://github.com/toltec-dev/toltec/blob/Eeems-patch-10/package/toltec-bootstrap/toltecctl#L885

But I'm not able to check the logic fully now.